src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIChatClient.cs

```diff
@@ -587,10 +589,9 @@ private sealed class OpenAIChatToolJson
                         string? result = resultContent.Result as string;
                         if (result is null && resultContent.Result is not null)
                         {
-                            JsonSerializerOptions options = ToolCallJsonSerializerOptions ?? JsonContext.Default.Options;
```

Comment:
Was the AOT app failing before these changes or is it just cleanup?

Reply:
It wasn't failing, but as I was reviewing the use I noticed this wouldn't do the desired thing.